transmogrifier/sources/marc.py

```diff
@@ -303,6 +385,16 @@ def get_optional_fields(self, xml: Tag) -> dict:
                 "kind": "Hierarchical Place Name",
             },
         ]
+        if fixed_length_data:
```

Comment:
I feel like there should be a less duplicative way to handle this but I couldn't come up with anything, let me know if you see an approach